samples/nrf9160/pdn/src/main.c

```diff
@@ -25,6 +29,65 @@ static const char * const event_str[] = {
 	[PDN_EVENT_IPV6_DOWN] = "IPv6 down",
 };
 
+static void ifaddrs_print(void)
+{
+	int err;
+
+	struct nrf_ifaddrs *ifaddrs;
+	struct nrf_ifaddrs *ifa;
+
+	char addr[NRF_INET6_ADDRSTRLEN] = {0};
+	char netmask[NRF_INET6_ADDRSTRLEN] = {0};
+	char broadaddr[NRF_INET6_ADDRSTRLEN] = {0};
+	char dstaddr[NRF_INET6_ADDRSTRLEN] = {0};
+
+	err = nrf_getifaddrs(&ifaddrs);
+	if (err) {
+		printk("nrf_getifaddrs() failed, err %d\n", err);
+		return;
+	}
+
+	printk("\nInterface addresses:\n");
+
+	ifa = ifaddrs;
+	while (ifa != NULL) {
+		switch (ifa->ifa_netmask->sa_family) {
+		case NRF_AF_INET:
+			nrf_inet_ntop(ifa->ifa_addr->sa_family,
+				      &SIN(ifa->ifa_addr)->sin_addr,
+				      addr, sizeof(addr));
+			break;
+		case NRF_AF_INET6:
+			nrf_inet_ntop(ifa->ifa_addr->sa_family,
+				      &SIN6(ifa->ifa_addr)->sin6_addr,
+				      addr, sizeof(addr));
+			break;
+		default:
+			printk("Unknown address family, %d\n", ifa->ifa_netmask->sa_family);
+		}
+
+		nrf_inet_ntop(ifa->ifa_netmask->sa_family,
```

Comment:
To be safe, you should probably check for both IPv4 and IPv6 to display Netmask.

You could wrap the conversion in a macro (or function) to make it easy.

Reply:
Example function (unless there is already a macro or something somewhere that does this):

```
static void snprintaddr(char *str, size_t size, struct nrf_sockaddr *addr)
{
	switch (addr->sa_family) {
		case NRF_AF_INET:
			nrf_inet_ntop(addr->sa_family,
				      &SIN(addr)->sin_addr,
				      str, size);
			break;
		case NRF_AF_INET6:
			nrf_inet_ntop(addr->sa_family,
				      &SIN6(addr)->sin6_addr,
				      str, size);
			break;
		default:
			snprintf(str, size, "Unknown family %d", addr->sa_family);
	}
}
```

Then, using it:

```
		snprintaddr(netmask, sizeof(netmask), ifa->ifa_netmask);
```

Reply:
The netmask will always be IPv4 from the modem, but I think adding the helper function is a good idea anyway, so I'll add it :)